### PR TITLE
Raise error when config file on command line is an empty string.

### PR DESCRIPTION
**Pull request by aloukina**
Added an extra check for empty config file name. This can generally only happen via the API.

### CODE REVIEW

File: skll/config.py [hunk 1]
```diff
@@ -236,6 +236,10 @@ def _parse_config_file(config_path):
     # Initialize logger
     logger = logging.getLogger(__name__)
 
+    # check that config_path is not empty
+    if config_path == "":
+        raise ValueError("The name of the configuration file is empty")
```
Comment (aoifecahill) [inline skll/config.py]: perhaps FileNotFoundError would be more appropriate?

**Comment by aoifecahill** [comment]
Adding more scope to this issue: what happens when the config file does not exists (different from it being blank)?

Could you also account for that case and add the relevant test?

**Comment by aloukina** [comment]
This is already addressed - SKLL raises IOError. The relevant test is in test_input line 130.

**Comment by aoifecahill** [comment]
Then maybe we should raise IOError here too?

**Comment by aoifecahill** [comment]
actually, if we are already checking for a file that doesn't exist, then I think this check belongs in the same place (``_setup_config_parse``)

**Comment by aloukina** [comment]
I thought of this but by that point the config_path is already converted to absolute path by realpath(). In case of empty filename this would be just the directory name which exists. So to catch the error we'll need to be checking whether supplied config_path is a directory which is a less transparent check.

**Comment by aoifecahill** [comment]
fair enough :)

**Comment by desilinguist** [comment]
So, you just need to change the `ValueError` to an `IOError`, sounds like?

**Comment by desilinguist** [comment]
Since, `coveralls.io` is still down. I ran this manually and the coverage was 91%. So, I am going to go ahead and merge this.

**Pull request merged**